pkg/reconciler/subscription/subscription.go

```diff
@@ -425,7 +439,13 @@ func (r *Reconciler) getChannel(ctx context.Context, sub *v1.Subscription) (*eve
 			return nil, fmt.Errorf("channel is not ready.")
 		}
 
-		statCh := duckv1.KReference{Name: channel.Status.Channel.Name, Namespace: sub.Namespace, Kind: channel.Status.Channel.Kind, APIVersion: channel.Status.Channel.APIVersion}
+		// if cross namespace referencing is enabled, namespace is updated to be the channel's namespace
+		if feature.FromContext(ctx).IsEnabled(feature.CrossNamespaceEventLinks) && sub.Spec.Channel.Namespace != "" {
+			channelNamespace = sub.Spec.Channel.Namespace
+		} else {
+			channelNamespace = sub.Namespace
+		}
```

Comment:
Here, the logic is in the correct order! Let's just try and use this same order for the tracker code above